src/arrayMethodReduce.js

```diff
@@ -5,7 +5,14 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let acc = arguments.length > 1 ? initialValue : this[0];
+    const start = arguments.length > 1 ? 0 : 1;
```

Comment:
do we need such complicated checks? we make the same thing in one `if` statement